src/SSTClosGovernorImp.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#ifndef SSTCLOSGOVERNORIMP_HPP_INCLUDE
+#define SSTCLOSGOVERNORIMP_HPP_INCLUDE
+
+#include "SSTClosGovernor.hpp"
+
+namespace geopm
+{
+    class SSTClosGovernorImp : public SSTClosGovernor
+    {
+        public:
+            SSTClosGovernorImp();
+            SSTClosGovernorImp(PlatformIO &platform_io, const PlatformTopo &platform_topo);
+            virtual ~SSTClosGovernorImp() = default;
+            void init_platform_io(void) override;
+            int clos_domain_type(void) const override;
+            void adjust_platform(const std::vector<double> &clos_by_core) override;
+            bool do_write_batch(void) const override;
+
+            // No-op if sst-tf is not supported. Mods both tf and cp
+            void enable_sst_turbo_prioritization() override;
+            void disable_sst_turbo_prioritization() override;
+
+        private:
+            PlatformIO &m_platform_io;
+            const PlatformTopo &m_platform_topo;
+            bool m_do_write_batch;
```

Comment:
Have you considered putting the larger data types to the top of the list, for making the object take up less space in memory? Perhaps we should rearrange the fields in such a way that there is less padding in between them, hence the object doesn't take up as much space in memory? We could save like 12 bytes worth of memory.

Reply:
Some member initializations depend on the outcome of other member initializations in this class. With this order, we can avoid initializing some members twice (once to create a throwaway instance, and again when we have the prerequisites ready).

This class typically gets one instance, so I don't think the 12B savings is a worthwhile trade-off of either initializing some of its members twice or adding more code complexity to work around it.